### PR TITLE
remove activate requirement from code base, skip splash if account created

### PR DESCRIPTION
**Pull request by YohayBar**
#### Main purpose:
remove activate requirement from code base, skip splash if account created
#### Technical description:
remove anything related to account activation - kept account create in auth but set to true to make sure there will be no update issue.
If account already created simply skip welcome page, in welcome page wait for account created plus animation ends to navigate  to market place

This change only works on playground environment right now as it based on server update that remove create account requirement as well.  This change will be merged only after Production is updated as well.

#### Tests:
Remove tests part related to activation, added test for animation ends before account created

### CODE REVIEW

File: sdk/src/test/java/com/kin/ecosystem/splash/presenter/SplashPresenterTest.java [hunk 1]
```diff
@@ -160,18 +116,39 @@ public void getStartedClicked_AccountNotCreated_ObserverOnChange_AccountCreated_
...
-		activateCapture.getValue().onResponse(null);
 		verify(splashView, times(0)).navigateToMarketPlace();
 
+		when(accountManager.isAccountCreated()).thenReturn(true);
 		accountStateObserver.getValue().onChanged(CREATION_COMPLETED);
+
 		verify(accountManager).removeAccountStateObserver(accountStateObserver.getValue());
 		verify(timer, times(2)).purge();
-		verify(splashView).navigateToMarketPlace();
+		verify(splashView, times(1)).navigateToMarketPlace();
```
Comment (OrenZak) [inline sdk/src/test/java/com/kin/ecosystem/splash/presenter/SplashPresenterTest.java]: Times 1 is the default

File: sdk/src/test/java/com/kin/ecosystem/splash/presenter/SplashPresenterTest.java [hunk 2]
```diff
@@ -160,18 +116,39 @@ public void getStartedClicked_AccountNotCreated_ObserverOnChange_AccountCreated_
...
+	@Test
+	public void getStartedClicked_AccountNotCreated_ObserverOnChange_AccountCreated_NavigateToMP_AnimationEnded() {
+		ArgumentCaptor<TimerTask> timeoutTask = ArgumentCaptor.forClass(TimerTask.class);
+		ArgumentCaptor<Observer<Integer>> accountStateObserver = ArgumentCaptor.forClass(Observer.class);
+		when(accountManager.isAccountCreated()).thenReturn(false);
+
+		splashPresenter.getStartedClicked();
+		verify(timer).schedule(timeoutTask.capture(), anyLong());
+		verify(accountManager).addAccountStateObserver(accountStateObserver.capture());
+		verify(splashView, times(1)).animateLoading();
```
Comment (OrenZak) [inline sdk/src/test/java/com/kin/ecosystem/splash/presenter/SplashPresenterTest.java]: Same

File: sdk/src/test/java/com/kin/ecosystem/splash/presenter/SplashPresenterTest.java [hunk 3]
```diff
@@ -160,18 +116,39 @@ public void getStartedClicked_AccountNotCreated_ObserverOnChange_AccountCreated_
...
+		when(accountManager.isAccountCreated()).thenReturn(true);
+		accountStateObserver.getValue().onChanged(CREATION_COMPLETED);
+
+		verify(accountManager).removeAccountStateObserver(accountStateObserver.getValue());
+		verify(timer, times(2)).purge();
+		verify(splashView, times(0)).navigateToMarketPlace();
+
+		splashPresenter.onAnimationEnded();
+		verify(splashView, times(1)).navigateToMarketPlace();
```
Comment (OrenZak) [inline sdk/src/test/java/com/kin/ecosystem/splash/presenter/SplashPresenterTest.java]: Same

File: sdk/src/main/java/com/kin/ecosystem/splash/presenter/SplashPresenter.java [hunk 4]
```diff
@@ -147,15 +130,7 @@ private void startCreationTimeout(final int sec) {
 	}
 
 	private void showTryAgainLater() {
-		if(shouldShowError) {
 			showToast(TRY_AGAIN);
```
[the remark this answers is not in the document]
Comment (YohayBar) [reply sdk/src/main/java/com/kin/ecosystem/splash/presenter/SplashPresenter.java]: not likely account observer need to trigger error simultaneously with timeout - but a good catch - I will make sure observer is removed before calling to showToast to 100% block such case

**Pull request merged**